### DISCUSSION TITLE
Inquiry Regarding Issues with ImgViewerPROGMEM on ESP32 RGB Display

### DISCUSSION THREAD

**Discussion by gilchi** [moononournation/Arduino_GFX#385] [Q&A] [answered]
Firstly, I appreciate the excellent library you provide. I am using the ESP32 RGB display (ESP32-8040S043), and I'm encountering an issue with ImgViewerPROGMEM. In version 1.3.4, images were displayed correctly, but in the latest version with the applied screen rotation option, there is a delay in loading images, and images appear partially broken. Could you please review if there is any issue with this?

gfx->draw16bitRGBBitmap(0, 0, (uint16_t*)kakao, 100, 100);

// In version 1.3.4 
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
    45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
    5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
    8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
    0 /* hsync_polarity */, 8 /* hsync_front_porch */, 4 /* hsync_pulse_width */, 8 /* hsync_back_porch */,
    0 /* vsync_polarity */, 8 /* vsync_front_porch */, 4 /* vsync_pulse_width */, 8 /* vsync_back_porch */,
    1 /* pclk_active_neg */, 16000000 /* prefer_speed */);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    800 /* width */, 480 /* height */, rgbpanel);

// In version 1.4.0
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
    45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
    5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
    8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
    0 /* hsync_polarity */, 8 /* hsync_front_porch */, 4 /* hsync_pulse_width */, 8 /* hsync_back_porch */,
    0 /* vsync_polarity */, 8 /* vsync_front_porch */, 4 /* vsync_pulse_width */, 8 /* vsync_back_porch */
);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    800 /* width */, 480 /* height */, rgbpanel, 2 /* rotation */, true /* auto_flush */);

**Comment by HaenZ33** [comment]
hi gilchi,
can you be more specific about the problem you are having/seening? I might have ran into the same problem.

**Reply by gilchi** [reply]
Hi haenZ33,
I have experienced that ImgViewerPROGMEM works fine in versions 1.3.4 and below. However, in higher versions, although the images are visible, some parts appear distorted, noisy, or partially broken.
Also, when running an example that loads images from an SD card in the latest version, I observed a similar issue.

**Comment by gilchi** [comment]
Additionally, in version 1.4.0, I found that using the "1 / pclk_active_neg /, 14000000 / prefer_speed" option improves noise and unstable screens. Furthermore, in version 1.4.0, applying screen rotation option 0 displays images correctly, but with rotation options 1, 2, and 3, small images of approximately 100X100 in size still appear distorted. 
Interestingly, there was no distortion in images with a size of 200X200 or larger. However, an issue with image jitter occurred with the "16000000 / prefer_speed" option. Please refer to the picture below.
![distor](https://github.com/moononournation/Arduino_GFX/assets/25917961/c63b9348-ab09-4fe8-9b2c-a5a1697a74b3)

**Reply by moononournation** [reply]
it may be a function bug, which function you are using to draw the bitmap?

**Reply by gilchi** [reply]
Thank you for the reply. I used the following function.
gfx->draw16bitRGBBitmap(0, 0, (uint16_t*)kakao, 100, 100);

**Reply by moononournation** [reply, marked as answer]
I think I may located the bug, please get update and try again.

**Reply by gilchi** [reply]
Thanks, it's working well now :)
